src/effects/deeplearning/DeepModelManager.cpp

```diff
@@ -0,0 +1,579 @@
+   (int httpCode, std::string body)
+   { 
+      if (!(httpCode == 200))
+      {
+         wxLogError(
+            wxString("GET request failed for url %s. Error code: %d")
+                  .Format(wxString(modelCardUrl), httpCode)
+         );
+      }
+      else
+      {
+         ModelCardHolder card(safenew ModelCard());
+         bool success = NewCardFromHuggingFace(card, body, repoID);
+         onCardFetched(success, card);
+      }
+   };
+
+   network_manager::ResponsePtr response = doGet(modelCardUrl, completionHandler);
+}
+
+void DeepModelManager::FetchModelSize(ModelCardHolder card, ModelSizeCallback onModelSizeRetrieved)
+{
+   if (card->IsLocal())
+   {
+      FilePath repoDir = GetRepoDir(card);
+      wxFileName modelPath = wxFileName(repoDir, "model.pt");
+
+      if (modelPath.FileExists())
+      {
+         size_t model_size = (size_t)wxFile(modelPath.GetFullPath(), wxFile::read).Length();
+
+         card->model_size(model_size);
+         onModelSizeRetrieved(model_size);
+      }
+   }
+   else
+   {
+      using namespace network_manager;
+
+      std::string modelUrl = GetRootURL(card->GetRepoID()) + "model.pt";
+
+      Request request(modelUrl);
+
+      NetworkManager &manager = NetworkManager::GetInstance();
+      ResponsePtr response = manager.doHead(request);
+
+      response->setRequestFinishedCallback(
+         [card, response, handler = std::move(onModelSizeRetrieved)](IResponse*)
+         {
+            if (!((response->getHTTPCode() == 200) || (response->getHTTPCode() == 302)))
+               return;
+
+            if (response->hasHeader("x-linked-size"))
+            {
+               std::string length = response->getHeader("x-linked-size");
+               std::cerr << length << std::endl;
+
+               size_t modelSize = std::stoi(length, nullptr, 10);
+               handler(modelSize);
+
+               card->model_size(modelSize);
+            }
+
+         }
+      );
+   }
+}
+
+ModelCardHolder DeepModelManager::GetEmptyCard()
+{
+   ModelCardHolder card(safenew ModelCard());
+   return card;
+}
+
+bool DeepModelManager::NewCardFromHuggingFace(ModelCardHolder card, const std::string &jsonBody, const std::string &repoID)
+{
+   bool success = true;
+
+   wxStringTokenizer st(wxString(repoID), wxT("/"));
+   std::string sAuthor = st.GetNextToken().ToStdString();
+   std::string sName = st.GetNextToken().ToStdString();
+   
+   try
+   {
+      DocHolder doc = parsers::ParseString(jsonBody);
+      card->Deserialize(doc, mModelCardSchema);
+      card->name(sName);
+      card->author(sAuthor);
+      card->SetLocal(false);
+      card->GetLocalPath(GetRepoDir(card).ToStdString());
+
+   }
+   catch (const InvalidModelCardDocument &e)
+   {
+      wxLogError(wxString(e.what()));
+      success = false;
+   }
+   catch (const char *msg)
+   { 
+      success = false;
+      wxLogError(wxString(msg));
+      wxASSERT(false);
```

Comment:
this block is run from a NetworkManager thread, when each card is retrieved from HuggingFace, so we need to catch any exceptions to avoid not throwing them in the main thread